x-pack/solutions/security/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/upgrade_with_preview.md

```diff
@@ -0,0 +1,706 @@
+- User should be able to upgrade a prebuilt rule that has some updates to [non-customizable fields](./shared_assets/non_customizable_rule_fields.md) in the latest version.
+- Any other fields that are not involved in the rule upgrade workflow, such as `enabled` or `exceptions`, should stay unchanged after rule upgrade.
+
+## Scenarios
+
+### Rule upgrade field preview
+
+#### Preview non-customized field that has an upgrade (AAB)
+
+**Automation**: Jest functional test for each \<field\>.
+
+```Gherkin
+Given an installed prebuilt rule
+And that rule has no customizations
+And <field> has an upgrade
+When user opens the Rule Update Flyout
+Then user should see <field> has no conflicts
+And <field> is shown collapsed
+When user expands <field>
+Then user should see <field> Diff View
+And user should see <field> Readonly View
+
+Examples:
+<field> = all customizable fields
+```
+
+#### Preview customized field that doesn't have an upgrade (ABA)
+
+**Automation**: Jest functional test for each \<field\>.
+
+```Gherkin
+Given an installed prebuilt rule
+And <field> is customized
+And <field> has no upgrades
+When user opens the Rule Update Flyout
+Then user should see <field> has a customized value
+And <field> has no conflicts
+And <field> is shown collapsed
+When user expands <field>
+Then user should see <field> Diff View
+And user should see <field> Readonly View
+
+Examples:
+<field> = all customizable fields
+```
+
+#### Preview customized field that has a matching upgrade (ABB)
+
+**Automation**: Jest functional test for each \<field\>.
+
+```Gherkin
+Given an installed prebuilt rule
+And <field> is customized
+And <field> has an upgrade matching customization
+When user opens the Rule Update Flyout
+Then user should see <field> has matching upgrade
+And <field> has no conflicts
+And <field> is shown collapsed
+When user expands <field>
+Then user should see <field> Diff View
+And user should see <field> Readonly View
+
+Examples:
+<field> = all customizable fields
+```
+
+#### Preview customized field that has an upgrade resulting in a solvable conflict (ABC, conflict solvable by diff algo)
+
+**Automation**: Jest functional test for each \<field\>.
+
+```Gherkin
+Given an installed prebuilt rule
+And <field> is customized
+And <field> has an upgrade resulting in a solvable conflict
+When user opens the Rule Update Flyout
+Then user should see <field> has a customized value
+And <field> has a conflict
+And <field> is shown expanded
+And <field> Diff View is shown
+And <field> Readonly View is shown
+And <field> Readonly View displays a merged value
+When user switches to edit form
+Then user should see <field> edit form
+And <field> edit form has merged value
+
+Examples: <field> whose diff algo supports values merging
+| data_source       |
+| tags              |
+| description       |
+| references        |
+| note              |
+| setup             |
+| threat_index      |
+| new_terms_fields  |
```

Comment:
In this test we should give concrete examples that would explain what solvable conflicts can look like, for a couple typical rule fields.

I'll address this in a follow-up PR.